power_grid_model_c/power_grid_model/include/power_grid_model/common/exception.hpp

```diff
@@ -112,6 +112,11 @@ class IterationDiverge : public PowerGridError {
     }
 };
 
+class MaxIterationReached : public PowerGridError {
```

Comment:
do we also want to update the `MathSolvers`?

Reply:
See discussion above.